services/alertmanager/alertmanager.go

```diff
@@ -40,22 +40,37 @@ const (
 	path                = "/srv/alertmanager/alertmanager.base.yml"
 )
 
-var resendInterval = 30 * time.Second
+// Environment variable to overwrite resend interval during testing
+const envResendInterval = "PERCONA_TEST_ALERTMANAGER_RESEND_INTERVAL"
+
+const defaultResendInterval = 30 * time.Second
 
 // Service is responsible for interactions with Alertmanager.
 type Service struct {
 	db *reform.DB
 	r  *Registry
 	l  *logrus.Entry
+
+	resendInterval time.Duration
 }
 
 // New creates new service.
 func New(db *reform.DB, alertsRegistry *Registry) *Service {
-	return &Service{
-		db: db,
-		r:  alertsRegistry,
-		l:  logrus.WithField("component", "alertmanager"),
+	l := logrus.WithField("component", "alertmanager")
+
+	s := &Service{
+		db:             db,
+		r:              alertsRegistry,
+		l:              l,
+		resendInterval: defaultResendInterval,
 	}
+
+	if d, err := time.ParseDuration(os.Getenv(envResendInterval)); err == nil && d > 0 {
+		s.resendInterval = d
+		l.Warnf("Resend interval changed to %s.", d)
+	}
```

Comment:
I think we want a resend interval update to affect both, alertmanager and registry.

Reply:
yup, that is why I moved resendInterval to registry and that way it can be accessed by both

Reply:
its in the latest commit that I pushed yesterday

Reply:
Great then)